scripts/scil_volume_pairwise_comparison.py

```diff
@@ -0,0 +1,206 @@
+                         'Please convert to uint8 (mask) or '
+                         'uint16 (labels).'.format(filename_2))
+
+    data_1 = get_data_as_labels(img_1)
+    data_2 = get_data_as_labels(img_2)
+
+    _, _, voxel_size, _ = get_reference_info(img_1)
+    voxel_size = np.product(voxel_size)
+
+    # Exclude 0 (background)
+    unique_values_1 = np.unique(data_1)[1:]
+    unique_values_2 = np.unique(data_2)[1:]
+    union_values = np.union1d(unique_values_1, unique_values_2)
+
+    dict_measures = {}
+    for val in union_values:
+        binary_1 = np.zeros(data_1.shape, dtype=np.uint8)
+        binary_1[data_1 == val] = 1
+        binary_2 = np.zeros(data_2.shape, dtype=np.uint8)
+        binary_2[data_2 == val] = 1
+
+        # These measures are in mm^3
+        volume_overlap = np.count_nonzero(binary_1 * binary_2)
+        volume_overreach = np.abs(np.count_nonzero(
+            binary_1 + binary_2) - volume_overlap)
+
+        if ratio:
+            count = np.count_nonzero(binary_1)
+            volume_overlap /= count
+            volume_overreach /= count
+
+        # These measures are in mm
+        bundle_adjacency_voxel = compute_bundle_adjacency_voxel(
+            binary_1, binary_2,
+            non_overlap=adjency_no_overlap)
+
+        # These measures are between 0 and 1
+        dice_vox, _ = compute_dice_voxel(binary_1,
+                                         binary_2)
+
+        measures_name = ['adjacency_voxels',
+                         'dice_voxels',
+                         'volume_overlap',
+                         'volume_overreach']
+
+        # If computing ratio, voxel size does not make sense
+        if ratio:
+            voxel_size = 1.
+        measures = [bundle_adjacency_voxel,
+                    dice_vox,
+                    volume_overlap * voxel_size,
+                    volume_overreach * voxel_size]
+
+        curr_dict = dict(zip(measures_name, measures))
+        for measure_name, measure in curr_dict.items():
+            if measure_name not in dict_measures:
+                dict_measures[measure_name] = {}
+            dict_measures[measure_name].update({int(val): float(measure)})
+
+    return dict_measures
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+    logging.getLogger().setLevel(logging.getLevelName(args.verbose))
+
+    assert_inputs_exist(parser, args.in_volumes, args.reference)
+    assert_outputs_exist(parser, args, [args.out_json])
+    assert_headers_compatible(parser, args.in_volumes,
+                              reference=args.reference)
+
+    if args.ratio and not args.single_compare:
+        parser.error('Can only compute ratio if also using `single_compare`')
+
+    nbr_cpu = validate_nbr_processes(parser, args)
+    if nbr_cpu > 1:
+        pool = multiprocessing.Pool(nbr_cpu)
+
+    if args.single_compare:
+        # Move the single_compare only once, at the end.
```

Comment:
This comment is not so clear for someone who doesn't know how the itertool.product works (me!)

`# Remove the single_compare from inputs and combine it to all other files.`